ecr/base.go

```diff
@@ -53,14 +53,26 @@ type ecrAPI interface {
 
 // getImageByDescriptor retrieves an image from ECR for a given OCI descriptor.
 func (b *ecrBase) getImageByDescriptor(ctx context.Context, desc ocispec.Descriptor) (*ecr.Image, error) {
+	// Only valid descriptors will be queried for.
+	if err := desc.Digest.Validate(); err != nil {
+		return nil, err
+	}
```

Comment:
I'd prefer to let the service do validation; it'll be able to tell us when a digest is invalid.

In general (and not specific to this change): client side validation is good on outputs, but can be a problem when new functionality is added on the remote side that allows previously-unsupported input to become supported.

Reply:
> I'd prefer to let the service do validation; it'll be able to tell us when a digest is invalid.

In this case, if the client is equipped with the associated digest algorithm then they are able to validate that the given hash is correctly specified. If the client *isn't* equipped with the required libraries then it'll fallback to acceptable format validation and thereby allow the service to support an unidentified hash type.

> In general (and not specific to this change): client side validation is good on outputs, but can be a problem when new functionality is added on the remote side that allows previously-unsupported input to become supported.

The supported types [and, most importantly here, their format](https://github.com/opencontainers/image-spec/blob/79b036d80240ae530a8de15e1d21c7ab9292c693/descriptor.md#digests) is specified by the [OCI image spec Descriptor section](https://github.com/opencontainers/image-spec/blob/79b036d80240ae530a8de15e1d21c7ab9292c693/descriptor.md) and I wouldn't expect them to change drastically anytime soon nor would I expect that service-side would allow vastly different kinds of hashes (at least, ones that would be usable in this context).

At this call site, I wouldn't expect there to be much of an opportunity to provide an invalid `digest.Digest` that containerd itself didn't already accept so I'm okay with removing the validation. However, I do think it's a simple, at-hand quick check that can be performed to give callers fast feedback regarding their input.

Reply:
Discussed offline, deferring this sort of check for now - such a check could always be added later if it proves to have value. 

The digest validation is removed in the latest push.